### PR TITLE
[libc++abi][test][VE] Add '-ldl' to support VE

### PR DESCRIPTION
**Pull request by kaz7**
Tests are linked under -nostdlib++ option, so it is required to add system's libraries correctly.  So, I add libdl.so which is required for VE.

### CODE REVIEW

**Comment by ldionne** [comment]
We don't want to start encoding target-specific information in our testing config files. If you need something specific for `ve`, the right way to do that would be to instead create a `ve-libc++abi-shared.cfg.in` file or something similar.

By the way, I've been seeing a few PRs about VE support lately. As documented [here](https://libcxx.llvm.org/#platform-and-compiler-support), our support policy requires adding a pre-commit CI bot to test every configuration that we claim to support. Is that something you can do for VE?

**Comment by kaz7** [comment]
> We don't want to start encoding target-specific information in our testing config files. If you need something specific for `ve`, the right way to do that would be to instead create a `ve-libc++abi-shared.cfg.in` file or something similar.

I see.  I'll do that.

> By the way, I've been seeing a few PRs about VE support lately. As documented [here](https://libcxx.llvm.org/#platform-and-compiler-support), our support policy requires adding a pre-commit CI bot to test every configuration that we claim to support. Is that something you can do for VE?

Oh, thank you for letting me know.  I didn't recognize that.  We have a buildbot and have a plan to add libcxx tests to the buildbot.  However, it is an after-commit buildbot.  Let me see what I can do.

**Comment by kaz7** [comment]
> > We don't want to start encoding target-specific information in our testing config files. If you need something specific for `ve`, the right way to do that would be to instead create a `ve-libc++abi-shared.cfg.in` file or something similar.
> I see. I'll do that.

I add following 2 configuration files for VE.  We will specifies them from LIBCXXABI_TEST_CONFIG and LIBCXX_TEST_CONFIG cmake variables.
```
 libcxx/test/configs/ve-libc++-shared.cfg.in       | 28 +++++++++++++++++++++++
 libcxxabi/test/configs/ve-libc++abi-shared.cfg.in | 28 +++++++++++++++++++++++
```

File: libcxx/test/configs/ve-libc++-shared.cfg.in [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This testing configuration handles running the test suite against LLVM's
```
Comment (ldionne) [inline libcxx/test/configs/ve-libc++-shared.cfg.in:1]: This should come with changes to `libcxx/utils/ci/run-buildbot` and `libcxx/utils/ci/buildkite-pipeline.yml` that allow running these tests in pre-commit. You can reach out to me on Discord privately and I'll help you with the BuildKite setup (agent tokens and so on).

**Review by ldionne — CHANGES_REQUESTED** [review]

File: libcxx/test/configs/ve-libc++-shared.cfg.in (hunk 1 above)
Comment (kaz7) [reply libcxx/test/configs/ve-libc++-shared.cfg.in:1]: Thanks.  I think I can use a machine runnning a buildbot for both purpose, buildbot and buildkite.  I'm discussing internally.

File: libcxx/test/configs/ve-libc++-shared.cfg.in [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+config.substitutions.append(('%{flags}',
+    '-pthread' + (' -isysroot {}'.format('@CMAKE_OSX_SYSROOT@') if '@CMAKE_OSX_SYSROOT@' else '')
+))
+config.substitutions.append(('%{compile_flags}',
+    '-nostdinc++ -I %{include} -I %{target-include} -I %{libcxx}/test/support'
+))
+# VE requires libdl library.
+config.substitutions.append(('%{link_flags}',
+    '-nostdlib++ -L %{lib} -Wl,-rpath,%{lib} -lc++ -ldl'
```
[the remark this answers is not in the document]
Comment (kaz7) [reply libcxx/test/configs/ve-libc++-shared.cfg.in:14]: In general, yes.  However, libcxx and libcxxabi want to link test programs under `-nostdlib++` option.  This avoid automatic linking.  Therefore we all need to specify all libraries like not only `-ldl` but also `-lc++`.

[the remark this answers is not in the document]
Comment (kaz7) [reply libcxx/test/configs/ve-libc++-shared.cfg.in:14]: I've noticed that It is possible to solve this problem by not modifying this cfg file but specifying cmake option like `-D_LIBCXX_TEST_COMPILER_CFLAGS="-ldl"`  Was that better for a solution of similar problems?  If so, I'd like to drop this ticket.  Thanks.

[the remark this answers is not in the document]
Comment (ldionne) [reply libcxx/test/configs/ve-libc++-shared.cfg.in:14]: IMO having a config file for `ve` is the right way to go if it has needs that are not met by the other config files. Basically, this should encode how to run a hello-world program on `ve`.

Although I don't think that's the right way to go anyway, I didn't find `_LIBCXX_TEST_COMPILER_CFLAGS` -- I'm still curious to understand what you meant there.

[the remark this answers is not in the document]
Comment (kaz7) [reply libcxx/test/configs/ve-libc++-shared.cfg.in:14]: Oh.  Sorry, I meant `RUNTIMES_ve-unknown-linux-gnu_LIBCXX_TEST_COMPILER_CFLAGS`.  I chopped it wrongly.  It's not a flag to specify library, but it is possible to use so.

Thank you for considering that.  I'm going to use above flag at this moment.  I'd like to be back with more common way to fix this.

**Comment by kaz7** [comment]
Closing this ticket once since it's better to have some common mechanisms to specify `-ldl` and other architecture dependent options without messing config file for each architecture.

**Pull request closed without merging**